core/src/main/java/org/mule/runtime/core/api/context/notification/ServerNotificationListener.java

```diff
@@ -6,11 +6,17 @@
  */
 package org.mule.runtime.core.api.context.notification;
 
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+
 /**
- * <code>ServerNotificationListener</code> is an observer interface that ojects can implement and register themselves with the
+ * <code>ServerNotificationListener</code> is an observer interface that objects can implement and register themselves with the
  * Mule Server to receive notifications when the server, model and components stop, start, initialise, etc.
  */
 public interface ServerNotificationListener<T extends ServerNotification> {
 
+  default ReactiveProcessor.ProcessingType getProcessingType() {
```

Comment:
Use IO as default until all notifications are non-blocking.  Also update javadoc of notificationListeners to say they should be non-blocking.